### PR TITLE
Minor tweaks to Windows build scripts

### PR DESCRIPTION
**Pull request by ixe013**
Fixes the issued mentionned in #3930. The build intructions now work as-is on Windows and produce a proper vault.exe binary

```
guill@PARALINT-04 C:\Users\guill\src\go\src\github.com\hashicorp\vault
> make bootstrap
github.com/mitchellh/gox (download)
github.com/kardianos/govendor (download)
github.com/client9/misspell (download)

guill@PARALINT-04 C:\Users\guill\src\go\src\github.com\hashicorp\vault
> make dev
==> Installing dependencies...
==> Cleaning old builds...
==> Preparing for development build...
==> Building...
Number of parallel builds: 7

-->   windows/amd64: github.com/hashicorp/vault

==> Results:

C:\Users\guill\src\go\src\github.com\hashicorp\vault\bin\vault.exe

guill@PARALINT-04 C:\Users\guill\src\go\src\github.com\hashicorp\vault
> C:\Users\guill\src\go\src\github.com\hashicorp\vault\bin\vault.exe --version
Vault v0.9.3 (d01be045d76b1483c49b4162df30c36989fc876b) (cgo)
```

### CODE REVIEW

File: make.bat [hunk 1]
```diff
@@ -6,7 +6,8 @@ set _EXITCODE=0
 REM If no target is provided, default to test.
 if [%1]==[] goto test
 
-set _TARGETS=bin,dev,generate,test,testacc,testrace,vet
+set _TARGETS=bin,bootstrap,dev,generate,test,testacc,testrace,vet
+set _EXTERNAL_TOOLS=github.com/mitchellh/gox,github.com/kardianos/govendor,github.com/client9/misspell/cmd/misspell
```
Comment (jefferai) [inline make.bat]: misspell shouldn't be necessary as a global thing for people to install

Comment (ixe013) [reply make.bat]: I agree about misspell... I replicated what was in `Makefile`.

```
$ grep misspell Makefile
        github.com/client9/misspell/cmd/misspell
        @misspell -error -source=text website/source
```

Your call whether we should keep it or not!

**Comment by ixe013** [comment]
Travis build failed, but I don't think it is related to my pull request though... Anything I should do?

File: make.bat (hunk 1 above)
Comment (jefferai) [reply make.bat]: It's only used for the website target, so not needed for normal builds.

**Comment by jefferai** [comment]
Nope, Travis likes to do that. We think it's some issue with DNS in the Travis containers since it only happens during race detection (which uses a different DNS stack).

**Comment by jefferai** [comment]
Thanks!

**Pull request merged**